components/addressmanager/src/lib.rs

```diff
@@ -55,124 +55,123 @@ impl AddressManager {
     }
 
     fn init_local_addresses(&mut self, tick_service: Arc<TickService>) -> Option<Extender> {
-        if let Some((net_addr, extend)) = self.configured_address() {
-            self.local_net_addresses.push(net_addr);
-
-            if let Some(ExtendHelper { gateway, local_addr, external_port }) = extend {
-                let gateway: igd_next::aio::Gateway<Tokio> = igd_next::aio::Gateway {
-                    addr: gateway.addr,
-                    root_url: gateway.root_url,
-                    control_url: gateway.control_url,
-                    control_schema_url: gateway.control_schema_url,
-                    control_schema: gateway.control_schema,
-                    provider: Tokio,
-                };
-                Some(Extender::new(
-                    tick_service,
-                    Duration::from_secs(UPNP_EXTEND_PERIOD),
-                    UPNP_DEADLINE_SEC,
-                    gateway,
-                    external_port,
-                    local_addr,
-                ))
-            } else {
-                None
-            }
+        self.local_net_addresses = self.local_addresses().collect();
+
+        let extender = if self.local_net_addresses.is_empty() && !self.config.disable_upnp {
+            let (net_address, ExtendHelper { gateway, local_addr, external_port }) = self.upnp()?;
```

Comment:
log in case of upnp error

Reply:
some of the exit paths within the `upnp` method already log an error, but not all

Reply:
it is already logged here:
https://github.com/biryukovmaxim/rusty-kaspa/blob/a76413f5b1ce6509ea6f0d8c63009af08c748e0b/components/addressmanager/src/lib.rs#L136
and here: https://github.com/biryukovmaxim/rusty-kaspa/blob/a76413f5b1ce6509ea6f0d8c63009af08c748e0b/components/addressmanager/src/lib.rs#L169
should I change it from warning to error?

Reply:
lines 133 and 134 are not logged

Reply:
btw line 169 `warn!("error adding port: {err}");` should start with a capital `Error` (keep it warn though). Also I think perhaps all upnp log messages should be prefixed with `[UPnP] `